frontend/src/app/reports/reports.service.ts

```diff
@@ -40,27 +69,98 @@ export class ReportsService {
         '3rd (2017)': 300,
         '4th (2017)': 400,
         '1st (2018)': 500,
-        '4th (2018)': 600
+        '2nd (2018)': 600
     };
 
-    constructor(private http: HttpClient, private configService: ConfigService) {
+    constructor(private http: HttpClient,
+                private configService: ConfigService,
+                private activatedRoute: ActivatedRoute) {
         this.config = configService.getConfig();
+        this.questionResolver = this.activatedRoute.snapshot.data.questionResolver;
     }
 
-    fetchReport(url: string) {
+    fetchReport(chart, grouping) {
+
         if (this.config.servicesReturnFakeData) {
-            return Observable.of(this.fetchFakeData(url)).delay(350);
+            return Observable.of(this.fetchFakeData(this.buildURL(chart, grouping))).delay(350);
         } else {
             const httpOptions = {
                 headers: new HttpHeaders({
                     'Content-Type': 'application/json',
                 })
             };
-            return this.http.get(url, httpOptions);
+            return this.http.get(this.buildURL(chart, grouping), httpOptions);
+        }
+    }
+
+    getCharts() {
+        // todo get all charts from backend
+        this.config.configQuestions.map(q => {
+            this.charts.push(
+                {value: 'Average compliance for ' + q.displayName,
+                    id: q.id,
+                    groupClusters: [{name: 'Time',
+                        disabled: false,
+                        groupings: [{value: 'year'},
+                            {value: 'quarter'},
+                            {value: 'month'},
+                            {value: 'week'}]},
+                        {name: 'Person',
+                            disabled: true,
+                            groupings: [{value: 'observer'}]},
+                        {name: 'Location',
+                            disabled: false,
+                            groupings: [{value: 'hospital'}]}]});
+            }
+        );
+        return this.charts;
+    }
+
+    /**
+        Dynamic (booleans):
+            average compliance by time:
+                /events/compliance/{question_id}/{timeframe}
+            average compliance by employee type:
+                /users/compliance/{question_id}
+            average compliance by location:
+                /compliance/{questionId}/location
+
+        Static:
+            number of observations by time:
+                /events/count/{time}
+                example: /events/count/quarter
+            number of observations by observer:
+                /events/count/observer
+     */
+    private buildURL(chart, grouping) {
+        if (chart.value.match('Number of observations')) {
```

Comment:
This function is hard to follow. Might try reducing the if/else branches, check this out on how: https://youtu.be/ldqDpmMkXgw or break chunks out into functions that define intent.